### ISSUE TITLE
Exception when trying to map SQLAlchemy ORM classes derived from a MappedAsDataclass-derived declarative base class

### ISSUE THREAD

**Issue by sh-at-cs** [dataclass-mapper/dataclass-mapper#29]
## Steps to reproduce

Consider:

```python
from dataclasses import dataclass

from dataclass_mapper import create_mapper
from sqlalchemy.orm import DeclarativeBase, Mapped, MappedAsDataclass, mapped_column

# Pure Python model

@dataclass
class A:
    id: str

# SQLAlchemy models

class Base(DeclarativeBase, MappedAsDataclass):
    pass

class OrmA(Base):
    __tablename__ = "a"

    id: Mapped[str] = mapped_column(primary_key=True)

# Mapper

create_mapper(A, OrmA)
```

This results in an exception:

```
TypeError: Field type 'sqlalchemy.orm.base.Mapped[str]' is not supported.
```

## Expected behavior

It should behave the same as when `Base` isn't derived from `MappedAsDataclass`, which is to generate a mapper without issues.

## Context

`MappedAsDataclass` is often (?) used because it allows static type checkers like Mypy or Pyright to type check arguments given to the derived classes' constructors. AFAIK, SQLAlchemy models not derived from a `MappedAsDataclass` base result in no constructor argument type checking whatsoever, which isn't ideal.

## Version info

Tested in version 2.0.0a1.

## Related issues

- #5

**Comment by jakobkogler** [comment]
Thanks, I'll look into it.
I never used `MappedAsDataclass` before.

**Comment by jakobkogler** [comment]
This was actually quite an easy feature to add.
The part of the code, that extracts all fields from an SQLAlchemy class already worked with `MappedAsDataclass` classes.

The only reason why it didn't work until now, is because it got detected as a normal dataclass, before the SQLAlchemy check ran. So the library believed that it didn't had anything to do with SQLAlchemy, and therefore didnt' understood the type annotations (`Mapped[...]`).

I've simply checked the order of the dataclass implementations, and first check if it is a SQLAlchemy model, and only if it's not one, I default to a Python dataclass.
=> Fixed in [Release v2.0.0a3](https://github.com/dataclass-mapper/dataclass-mapper/releases/tag/v2.0.0a3)
